mergify_engine/check_api.py

```diff
@@ -261,11 +267,23 @@ async def set_check_run(
             github_types.ISODateTimeType, ended_at
         )
 
-    checks = sorted(
-        (c for c in await ctxt.pull_engine_check_runs if c["name"] == name),
-        key=lambda c: c["id"],
-        reverse=True,
-    )
+    if ctxt._caches.pull_check_runs.get() is cache.Unset:
+        checks = sorted(
+            await get_checks_for_ref(
+                ctxt,
+                ctxt.pull["head"]["sha"],
+                check_name=name,
+                app_id=config.INTEGRATION_ID,
+            ),
+            key=lambda c: c["id"],
+            reverse=True,
+        )
```

Comment:
I'm not a huge fan either. The thing is that if we reach this without the cached loaded, it means it can't be loaded all checks anyways.

pull_check_runs contains all check runs not only our check runs. Here, we are looking for the id of one of our check runs to update it. So if the cache is not there we try to get the id manually.

I will try to split `set_check_api()` to make all of this clear, not sure it's easy.

Reply:
I finally moved in the `Context()` the choice to use or not the check runs cache